app/lib/backend_api_logic/service_extension.rb

```diff
@@ -8,7 +8,9 @@ module ServiceExtension
       has_many :backend_api_configs, inverse_of: :service, dependent: :destroy
       has_many :backend_apis, through: :backend_api_configs
 
-      alias_method :backend_api, :first_backend_api
+      delegate :backend_api, to: :backend_api_proxy
+      delegate :api_backend, to: :backend_api, allow_nil: true
```

Comment:
I do not get it, it can't be nil that is the whole point of the backend_api_proxy